### PR TITLE
Protocol spec changes

### PR DESCRIPTION
**Pull request by jiyoontbd**
blocked by https://github.com/TBD54566975/tbdex/pull/293 merging

### CODE REVIEW

File: tbdex
Comment (jiyoontbd) [inline tbdex:1]: this is what's latest in tbdex main https://github.com/TBD54566975/tbdex/commit/6af20c9d52a1497bef72ee88f712496844edd372

**Comment by jiyoontbd** [comment]
ci will fail - to be addressed by https://github.com/TBD54566975/tbdex-kt/issues/221

File: protocol/src/test/kotlin/tbdex/sdk/protocol/TbdexTestVectorsProtocol.kt [hunk 1]
```diff
@@ -23,36 +24,36 @@ class TbdexTestVectorsProtocol {
...
   @Test
   fun parse_quote() {
     val vector = TestVectors.getVector("parse-quote.json")
     assertNotNull(vector)
-    testNonErrorMessageTestVector<Quote>(vector)
+    testSuccessMessageTestVector<Quote>(vector)
   }
 
   @Test
   fun parse_rfq() {
```
Comment (diehuxx) [inline protocol/src/test/kotlin/tbdex/sdk/protocol/TbdexTestVectorsProtocol.kt:53]: For the benefit of other reviews: This is one of the two tests that we expect to fail for this PR. The only other test that should fail is `ValidatorTest > validateFailsWithInvalidRfqData()`. All other tests should (and do) pass.

 Context: #221

**Review by diehuxx — APPROVED** [review]
Great work! Slick and polished. I pinky promise to fix the two broken tests in this follow up PR #222  very soon

**Review by mistermoe — APPROVED** [review]
Nice work @jiyoontbd !

**Pull request merged**